client/src/services/event/event.service.js

```diff
@@ -0,0 +1,32 @@
+export default class EventService {
+  constructor($http) {
+    this.$inject = ['$http'];
+    this.$http = $http;
+    this.events = [];
+  }
+
+  loadEventsByPlanId(planId) {
+    return this.$http.get(`/api/events/${planId}`).then(response => this.events = response.data);
+  }
+
+  getEvent(eventId) { return this.events.find(event => event._id === eventId); }
+
+  submitNewEvent(event) { return this.$http.post('/api/event', event); }
+
+  updateEvent(eventId, updatedEvent) {
+    if (typeof eventId === 'object' && updatedEvent === undefined) {

```

Comment:
what if they provide both? error out?

Reply:
no, it's fine if they do.
Running it by Dom